diff --git a/src/configure b/src/configure
index 10d0ff5..7065417 100755
--- a/src/configure
+++ b/src/configure
@@ -34,16 +34,16 @@ esac
 eval "$("$src_dir"/shobj-conf -C "$CC" -o "$host_os")"
 
 sed "
-  s,@CC@,${CC},
-  s,@CFLAGS@,${CFLAGS},
-  s,@LOCAL_CFLAGS@,${LOCAL_CFLAGS},
-  s,@DEFS@,${DEFS},
-  s,@LOCAL_DEFS@,${LOCAL_DEFS},
-  s,@SHOBJ_CC@,${SHOBJ_CC},
-  s,@SHOBJ_CFLAGS@,${SHOBJ_CFLAGS},
-  s,@SHOBJ_LD@,${SHOBJ_LD},
-  s,@SHOBJ_LDFLAGS@,${SHOBJ_LDFLAGS//,/\\,},
-  s,@SHOBJ_XLDFLAGS@,${SHOBJ_XLDFLAGS//,/\\,},
-  s,@SHOBJ_LIBS@,${SHOBJ_LIBS},
-  s,@SHOBJ_STATUS@,${SHOBJ_STATUS},
+  s#@CC@#${CC}#
+  s#@CFLAGS@#${CFLAGS}#
+  s#@LOCAL_CFLAGS@#${LOCAL_CFLAGS}#
+  s#@DEFS@#${DEFS}#
+  s#@LOCAL_DEFS@#${LOCAL_DEFS}#
+  s#@SHOBJ_CC@#${SHOBJ_CC}#
+  s#@SHOBJ_CFLAGS@#${SHOBJ_CFLAGS}#
+  s#@SHOBJ_LD@#${SHOBJ_LD}#
+  s#@SHOBJ_LDFLAGS@#${SHOBJ_LDFLAGS//,/\\,}#
+  s#@SHOBJ_XLDFLAGS@#${SHOBJ_XLDFLAGS//,/\\,}#
+  s#@SHOBJ_LIBS@#${SHOBJ_LIBS}#
+  s#@SHOBJ_STATUS@#${SHOBJ_STATUS}#
 " "$src_dir"/Makefile.in > "$src_dir"/Makefile
