diff -urN netcdf-4.4.1.1-orig/nc-config.in netcdf-4.4.1.1/nc-config.in
--- netcdf-4.4.1.1-orig/nc-config.in	2017-03-21 07:58:00.933833160 -0400
+++ netcdf-4.4.1.1/nc-config.in	2017-03-21 08:04:06.000000000 -0400
@@ -6,12 +6,10 @@
 
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-libdir=@libdir@
 includedir=@includedir@
 
 cc="@CC@"
 cflags="-I${includedir} @CPPFLAGS@"
-libs="-L${libdir} @NC_LIBS@"
 
 has_dap="@HAS_DAP@"
 has_nc2="@HAS_NC2@"
@@ -219,7 +217,7 @@
        	;;
 
      --libs)
-       	echo $libs
+       	pkg-config netcdf --libs
        	;;
 
     --prefix)
