diff -urN 4.0.1.orig/ui/include/classes/core/CConfigFile.php 4.0.1/ui/include/classes/core/CConfigFile.php
--- 4.0.1.orig/ui/include/classes/core/CConfigFile.php	2018-10-29 19:00:25.270221980 +0200
+++ 4.0.1/ui/include/classes/core/CConfigFile.php	2018-10-29 19:00:54.666586770 +0200
@@ -25,7 +25,7 @@
 	const CONFIG_ERROR = 2;
 	const CONFIG_VAULT_ERROR = 3;
 
-	const CONFIG_FILE_PATH = '/conf/zabbix.conf.php';
+	const CONFIG_FILE_PATH = '/etc/zabbix/web/zabbix.conf.php';
 
 	private static $supported_db_types = [
 		ZBX_DB_MYSQL => true,
diff -urN 4.0.1.orig/ui/include/classes/core/ZBase.php 4.0.1/ui/include/classes/core/ZBase.php
--- 4.0.1.orig/ui/include/classes/core/ZBase.php	2018-10-29 19:00:25.270221980 +0200
+++ 4.0.1/ui/include/classes/core/ZBase.php	2018-10-29 19:00:36.450360730 +0200
@@ -320,7 +320,7 @@
 	 * @throws Exception
 	 */
 	protected function setMaintenanceMode() {
-		require_once 'conf/maintenance.inc.php';
+		require_once '/etc/zabbix/web/maintenance.inc.php';
 
 		if (defined('ZBX_DENY_GUI_ACCESS')) {
			if (!isset($ZBX_GUI_ACCESS_IP_RANGE) || !in_array(CWebUser::getIp(), $ZBX_GUI_ACCESS_IP_RANGE)) {
@@ -289,7 +289,7 @@
 	 * Load zabbix config file.
 	 */
 	protected function loadConfigFile() {
-		$configFile = $this->getRootDir().CConfigFile::CONFIG_FILE_PATH;
+		$configFile = CConfigFile::CONFIG_FILE_PATH;
 		$config = new CConfigFile($configFile);
 		$this->config = $config->load();
 	}
diff -urN 4.0.1.orig/ui/include/classes/setup/CSetupWizard.php 4.0.1/ui/include/classes/setup/CSetupWizard.php
--- 4.0.1.orig/ui/include/classes/setup/CSetupWizard.php	2018-10-29 19:00:25.274222030 +0200
+++ 4.0.1/ui/include/classes/setup/CSetupWizard.php	2018-10-29 19:00:36.450360730 +0200
@@ -336,7 +336,7 @@

 		$this->setConfig('ZBX_CONFIG_FILE_CORRECT', true);
 
-		$config_file_name = APP::getInstance()->getRootDir().CConfigFile::CONFIG_FILE_PATH;
+		$config_file_name = CConfigFile::CONFIG_FILE_PATH;
 		$config = new CConfigFile($config_file_name);
 		$config->config = [
 			'DB' => [
@@ -507,7 +507,7 @@
 				// make zabbix.conf.php downloadable
 				header('Content-Type: application/x-httpd-php');
 				header('Content-Disposition: attachment; filename="'.basename(CConfigFile::CONFIG_FILE_PATH).'"');
-				$config = new CConfigFile(APP::getInstance()->getRootDir().CConfigFile::CONFIG_FILE_PATH);
+				$config = new CConfigFile(CConfigFile::CONFIG_FILE_PATH);
 				$config->config = [
 					'DB' => [
 						'TYPE' => $this->getConfig('DB_TYPE'),
