diff -urN 4.0.1.orig/frontends/php/include/defines.inc.php 4.0.1/frontends/php/include/defines.inc.php
--- 4.0.1.orig/frontends/php/include/defines.inc.php	2018-10-29 19:17:06.402752080 +0200
+++ 4.0.1/frontends/php/include/defines.inc.php	2018-10-29 19:18:01.187444598 +0200
@@ -62,7 +62,7 @@
 define('ZBX_WIDGET_ROWS', 20);
 
 define('ZBX_FONTPATH',				realpath('fonts')); // where to search for font (GD > 2.0.18)
-define('ZBX_GRAPH_FONT_NAME',		'DejaVuSans'); // font file name
+define('ZBX_GRAPH_FONT_NAME',		'graphfont'); // font file name
 define('ZBX_GRAPH_LEGEND_HEIGHT',	120); // when graph height is less then this value, some legend will not show up
 
 define('ZBX_SCRIPT_TIMEOUT',		60); // in seconds
@@ -107,7 +107,7 @@
 define('ZBX_ACK_STS_WITH_UNACK',		2);
 define('ZBX_ACK_STS_WITH_LAST_UNACK',	3);
 
-define('ZBX_FONT_NAME', 'DejaVuSans');
+define('ZBX_FONT_NAME', 'graphfont');
 
 define('ZBX_AUTH_INTERNAL',	0);
 define('ZBX_AUTH_LDAP',		1);
