diff -urN zabbix-4.4.1-orig/frontends/php/include/defines.inc.php zabbix-4.4.1/frontends/php/include/defines.inc.php
--- zabbix-4.4.1-orig/frontends/php/include/defines.inc.php	2019-11-08 18:10:48.000000000 +0300
+++ zabbix-4.4.1/frontends/php/include/defines.inc.php	2019-11-08 18:12:02.000000000 +0300
@@ -64,7 +64,7 @@
 define('ELASTICSEARCH_RESPONSE_DOCUMENTS',		2);
 
 define('ZBX_FONTPATH',				realpath('assets/fonts')); // where to search for font (GD > 2.0.18)
-define('ZBX_GRAPH_FONT_NAME',		'DejaVuSans'); // font file name
+define('ZBX_GRAPH_FONT_NAME',		'graphfont'); // font file name
 define('ZBX_GRAPH_LEGEND_HEIGHT',	120); // when graph height is less then this value, some legend will not show up
 
 define('ZBX_SCRIPT_TIMEOUT',		60); // in seconds
@@ -108,7 +108,7 @@
 define('TRIGGERS_OPTION_ALL',				2);
 define('TRIGGERS_OPTION_IN_PROBLEM',		3);
 
-define('ZBX_FONT_NAME', 'DejaVuSans');
+define('ZBX_FONT_NAME', 'graphfont');
 
 define('ZBX_AUTH_INTERNAL',	0);
 define('ZBX_AUTH_LDAP',		1);
