diff -Nru zabbix-3.0.2.orig/src/libs/zbxicmpping/icmpping.c zabbix-3.0.2/src/libs/zbxicmpping/icmpping.c
--- zabbix-3.0.2.orig/src/libs/zbxicmpping/icmpping.c	2016-04-20 18:51:21.000000000 +0900
+++ zabbix-3.0.2/src/libs/zbxicmpping/icmpping.c	2016-04-20 22:28:59.000000000 +0900
@@ -42,34 +42,7 @@
 
 static void	get_source_ip_option(const char *fping, const char **option, unsigned char *checked)
 {
-	FILE	*f;
-	char	*p, tmp[MAX_STRING_LEN];
-
-	zbx_snprintf(tmp, sizeof(tmp), "%s -h 2>&1", fping);
-
-	if (NULL == (f = popen(tmp, "r")))
-		return;
-
-	while (NULL != fgets(tmp, sizeof(tmp), f))
-	{
-		for (p = tmp; isspace(*p); p++)
-			;
-
-		if ('-' == p[0] && 'I' == p[1] && isspace(p[2]))
-		{
-			*option = "-I";
-			continue;
-		}
-
-		if ('-' == p[0] && 'S' == p[1] && isspace(p[2]))
-		{
-			*option = "-S";
-			break;
-		}
-	}
-
-	pclose(f);
-
+	*option = "-S";
 	*checked = 1;
 }
 
