diff -urN postgis-1.5.8-orig/postgis/lwgeom_accum.c postgis-1.5.8/postgis/lwgeom_accum.c
--- postgis-1.5.8-orig/postgis/lwgeom_accum.c	2016-05-22 16:59:04.033580517 -0400
+++ postgis-1.5.8/postgis/lwgeom_accum.c	2016-05-22 17:24:48.000000000 -0400
@@ -103,21 +103,23 @@
 		        (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
 		         errmsg("could not determine input data type")));
 
+#if POSTGIS_PGSQL_VERSION == 84
 	if (fcinfo->context && IsA(fcinfo->context, AggState))
+	{
 		aggcontext = ((AggState *) fcinfo->context)->aggcontext;
-#if POSTGIS_PGSQL_VERSION == 84
+	}
 	else if (fcinfo->context && IsA(fcinfo->context, WindowAggState))
+	{
 		aggcontext = ((WindowAggState *) fcinfo->context)->wincontext;
-#endif
-#if POSTGIS_PGSQL_VERSION > 84
-	else if (fcinfo->context && IsA(fcinfo->context, WindowAggState))
-		aggcontext = ((WindowAggState *) fcinfo->context)->aggcontext;
-#endif
+	}
 	else
+#else
+	if ( ! AggCheckCallContext(fcinfo, &aggcontext) )
+#endif
 	{
 		/* cannot be called directly because of dummy-type argument */
 		elog(ERROR, "array_agg_transfn called in non-aggregate context");
-		aggcontext = NULL;		/* keep compiler quiet */
+		aggcontext = NULL;  /* keep compiler quiet */
 	}
 
 	if ( PG_ARGISNULL(0) )
