diff -urN postgis-1.5.8-orig/postgis/geography_estimate.c postgis-1.5.8/postgis/geography_estimate.c
--- postgis-1.5.8-orig/postgis/geography_estimate.c	2015-06-27 14:25:14.585005141 +0200
+++ postgis-1.5.8/postgis/geography_estimate.c	2015-06-27 14:26:40.000000000 +0200
@@ -22,6 +22,10 @@
 #include "libgeom.h"
 #include "lwgeom_pg.h"
 
+#if POSTGIS_PGSQL_VERSION >= 93
+  #include "access/htup_details.h"
+#endif
+
 /* Prototypes */
 Datum geography_gist_selectivity(PG_FUNCTION_ARGS);
 Datum geography_gist_join_selectivity(PG_FUNCTION_ARGS);
