diff -Naupr p7zip_9.20.1.orig/CPP/7zip/Bundles/Format7zFree/makefile p7zip_9.20.1/CPP/7zip/Bundles/Format7zFree/makefile
--- p7zip_9.20.1.orig/CPP/7zip/Bundles/Format7zFree/makefile	2010-11-07 16:41:43.000000000 +0100
+++ p7zip_9.20.1/CPP/7zip/Bundles/Format7zFree/makefile	2011-07-26 14:11:12.480236487 +0200
@@ -158,14 +158,6 @@ NSIS_OBJS = \
   NsisIn.o \
   NsisRegister.o \
 
-RAR_OBJS = \
-  RarHandler.o \
-  RarHeader.o \
-  RarIn.o \
-  RarItem.o \
-  RarVolumeInStream.o \
-  RarRegister.o \
-
 TAR_OBJS = \
   TarHandler.o \
   TarHandlerOut.o \
@@ -243,13 +235,6 @@ COMPRESS_OBJS = \
   ZlibEncoder.o \
   ZDecoder.o \
 
-COMPRESS_OBJS_NON_FREE = \
-  Rar1Decoder.o \
-  Rar2Decoder.o \
-  Rar3Decoder.o \
-  Rar3Vm.o \
-  RarCodecsRegister.o \
-
 CRYPTO_OBJS = \
   7zAes.o \
   7zAesRegister.o \
@@ -257,8 +242,6 @@ CRYPTO_OBJS = \
   MyAes.o \
   Pbkdf2HmacSha1.o \
   RandGen.o \
-  Rar20Crypto.o \
-  RarAes.o \
   Sha1.o \
   WzAes.o \
   ZipCrypto.o \
@@ -314,7 +297,6 @@ OBJS = \
   $(HFS_OBJS) \
   $(ISO_OBJS) \
   $(NSIS_OBJS) \
-  $(RAR_OBJS) \
   $(TAR_OBJS) \
   $(UDF_OBJS) \
   $(WIM_OBJS) \
diff -Naupr p7zip_9.20.1.orig/CPP/7zip/Bundles/Format7zFree/makefile.list p7zip_9.20.1/CPP/7zip/Bundles/Format7zFree/makefile.list
--- p7zip_9.20.1.orig/CPP/7zip/Bundles/Format7zFree/makefile.list	2010-11-07 16:42:33.000000000 +0100
+++ p7zip_9.20.1/CPP/7zip/Bundles/Format7zFree/makefile.list	2011-07-26 14:11:31.068174008 +0200
@@ -133,13 +133,6 @@ SRCS=\
  ../../Archive/Nsis/NsisIn.cpp \
  ../../Archive/Nsis/NsisRegister.cpp \
  \
- ../../Archive/Rar/RarHandler.cpp \
- ../../Archive/Rar/RarHeader.cpp \
- ../../Archive/Rar/RarIn.cpp \
- ../../Archive/Rar/RarItem.cpp \
- ../../Archive/Rar/RarVolumeInStream.cpp \
- ../../Archive/Rar/RarRegister.cpp \
- \
  ../../Archive/Tar/TarHandler.cpp \
  ../../Archive/Tar/TarHandlerOut.cpp \
  ../../Archive/Tar/TarHeader.cpp \
@@ -220,8 +213,6 @@ SRCS=\
  ../../Crypto/RandGen.cpp \
  ../../Crypto/Sha1.cpp \
  ../../Crypto/WzAes.cpp \
- ../../Crypto/Rar20Crypto.cpp \
- ../../Crypto/RarAes.cpp \
  ../../Crypto/ZipCrypto.cpp \
  ../../Crypto/ZipStrong.cpp
 
diff -Naupr p7zip_9.20.1.orig/makefile p7zip_9.20.1/makefile
--- p7zip_9.20.1.orig/makefile	2011-03-13 12:52:45.000000000 +0100
+++ p7zip_9.20.1/makefile	2011-07-26 14:11:43.546132132 +0200
@@ -56,7 +56,6 @@ depend:
 	$(MAKE) -C CPP/7zip/UI/Client7z           depend
 	$(MAKE) -C CPP/7zip/UI/Console            depend
 	$(MAKE) -C CPP/7zip/Bundles/Format7zFree  depend
-	$(MAKE) -C CPP/7zip/Compress/Rar          depend
 	$(MAKE) -C CPP/7zip/UI/GUI                depend
 	$(MAKE) -C CPP/7zip/UI/FileManager        depend
 	$(MAKE) -C check/my_86_filter             depend
@@ -68,7 +67,6 @@ sfx: common
 common7z:common
 	$(MKDIR) bin/Codecs
 	$(MAKE) -C CPP/7zip/Bundles/Format7zFree all
-	$(MAKE) -C CPP/7zip/Compress/Rar         all
 
 7z: common7z
 	$(MAKE) -C CPP/7zip/UI/Console           all
@@ -93,7 +91,6 @@ clean:
 	$(MAKE) -C CPP/7zip/UI/FileManager       clean
 	$(MAKE) -C CPP/7zip/UI/GUI               clean
 	$(MAKE) -C CPP/7zip/Bundles/Format7zFree clean
-	$(MAKE) -C CPP/7zip/Compress/Rar         clean
 	$(MAKE) -C CPP/7zip/Compress/LZMA_Alone  clean
 	$(MAKE) -C CPP/7zip/Bundles/AloneGCOV    clean
 	$(MAKE) -C CPP/7zip/TEST/TestUI          clean
