diff -urN faad2-2.8.8-orig/common/mp4ff/Makefile.am faad2-2.8.8/common/mp4ff/Makefile.am
--- faad2-2.8.8-orig/common/mp4ff/Makefile.am	2017-07-17 15:04:02.000000000 +0300
+++ faad2-2.8.8/common/mp4ff/Makefile.am	2019-12-13 16:09:15.000000000 +0300
@@ -1,5 +1,5 @@
 noinst_LIBRARIES = libmp4ff.a
-noinst_HEADERS = mp4ff.h mp4ffint.h
+noinst_HEADERS = mp4ff.h mp4ff_int_types.h mp4ffint.h
 
 libmp4ff_a_CFLAGS = -DUSE_TAGGING=1
 
