From c35ec59b8a314aa030c7c09a4bd66e70834c4bb3 Mon Sep 17 00:00:00 2001
From: Ryan Davis <ryand-ruby@zenspider.com>
Date: Sun, 1 Jan 2017 22:06:39 -0800
Subject: [PATCH] Fixed `rbenv init -` output to work w/ no args and bash's
 `set -u`.

---
 libexec/rbenv-init | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libexec/rbenv-init b/libexec/rbenv-init
index ac9d4cf..dffbec3 100755
--- a/libexec/rbenv-init
+++ b/libexec/rbenv-init
@@ -138,7 +138,7 @@ esac
 if [ "$shell" != "fish" ]; then
 IFS="|"
 cat <<EOS
-  command="\$1"
+  command="\${1:-}"
   if [ "\$#" -gt 0 ]; then
     shift
   fi
