diff -urN rbenv-1.2.0-orig/libexec/rbenv-versions rbenv-1.2.0/libexec/rbenv-versions
--- rbenv-1.2.0-orig/libexec/rbenv-versions	2021-09-29 21:47:10.000000000 +0300
+++ rbenv-1.2.0/libexec/rbenv-versions	2022-08-19 13:03:39.000000000 +0300
@@ -102,7 +102,7 @@
 current_version="$(rbenv-version-name || true)"
 while read -r version; do
   if [ "$version" == "$current_version" ]; then
-    echo "* $(rbenv-version 2>/dev/null)"
+    echo "→ $(rbenv-version 2>/dev/null)"
   else
     echo "  $version"
   fi
