diff --color -urN rbenv-1.3.0-orig/libexec/rbenv-versions rbenv-1.3.0-arrow/libexec/rbenv-versions
--- rbenv-1.3.0-orig/libexec/rbenv-versions	2024-07-05 16:02:25.000000000 +0300
+++ rbenv-1.3.0-arrow/libexec/rbenv-versions	2024-09-06 01:32:53.000000000 +0300
@@ -76,7 +76,7 @@
 current_version="$(rbenv-version-name || true)"
 while read -r version; do
   if [ "$version" == "$current_version" ]; then
-    echo "* $(rbenv-version 2>/dev/null)"
+    echo "→ $(rbenv-version 2>/dev/null)"
   else
     echo "  $version"
   fi
