diff -urN gradle-6.0.1-orig/build.gradle.kts gradle-6.0.1/build.gradle.kts
--- gradle-6.0.1-orig/build.gradle.kts	1980-02-01 00:00:00.000000000 +0300
+++ gradle-6.0.1/build.gradle.kts	2019-12-16 17:07:05.000000000 +0300
@@ -18,7 +18,6 @@
 import org.gradle.build.Install
 import org.gradle.gradlebuild.ProjectGroups.implementationPluginProjects
 import org.gradle.gradlebuild.ProjectGroups.javaProjects
-import org.gradle.gradlebuild.ProjectGroups.kotlinJsProjects
 import org.gradle.gradlebuild.ProjectGroups.pluginProjects
 import org.gradle.gradlebuild.ProjectGroups.publicJavaProjects
 import org.gradle.gradlebuild.UpdateBranchStatus
@@ -158,14 +157,6 @@
             name = "Gradle libs"
             url = uri("https://repo.gradle.org/gradle/libs")
         }
-        maven {
-            name = "kotlinx"
-            url = uri("https://kotlin.bintray.com/kotlinx/")
-        }
-        maven {
-            name = "kotlin-eap"
-            url = uri("https://dl.bintray.com/kotlin/kotlin-eap")
-        }
     }
 
     // patchExternalModules lives in the root project - we need to activate normalization there, too.
@@ -337,7 +328,6 @@
     coreRuntime(project(":runtimeApiInfo"))
     coreRuntime(project(":wrapper"))
     coreRuntime(project(":installationBeacon"))
-    coreRuntime(project(":kotlinDsl"))
 
     pluginProjects.forEach { gradlePlugins(it) }
     implementationPluginProjects.forEach { gradlePlugins(it) }
@@ -350,8 +340,6 @@
     coreRuntimeExtensions(project(":instantExecution"))
     coreRuntimeExtensions(project(":pluginUse"))
     coreRuntimeExtensions(project(":workers"))
-    coreRuntimeExtensions(project(":kotlinDslProviderPlugins"))
-    coreRuntimeExtensions(project(":kotlinDslToolingBuilders"))
 
     testRuntime(project(":apiMetadata"))
 }
diff -urN gradle-6.0.1-orig/buildSrc/build.gradle.kts gradle-6.0.1/buildSrc/build.gradle.kts
--- gradle-6.0.1-orig/buildSrc/build.gradle.kts	1980-02-01 00:00:00.000000000 +0300
+++ gradle-6.0.1/buildSrc/build.gradle.kts	2019-12-16 17:15:02.000000000 +0300
@@ -14,15 +14,12 @@
  * limitations under the License.
  */
 
-import org.gradle.kotlin.dsl.plugins.dsl.KotlinDslPlugin
 
 import java.io.File
 import java.util.Properties
 
 plugins {
     `java`
-    `kotlin-dsl` apply false
-    id("org.gradle.kotlin-dsl.ktlint-convention") version "0.4.1" apply false
 }
 
 subprojects {
@@ -34,10 +31,6 @@
             applyGroovyProjectConventions()
         }
 
-        if (file("src/main/kotlin").isDirectory || file("src/test/kotlin").isDirectory) {
-            applyKotlinProjectConventions()
-        }
-
         java {
             sourceCompatibility = JavaVersion.VERSION_1_8
             targetCompatibility = JavaVersion.VERSION_1_8
@@ -82,14 +75,6 @@
             name = "Gradle snapshot libs"
             url = uri("https://repo.gradle.org/gradle/libs-snapshots")
         }
-        maven {
-            name = "kotlinx"
-            url = uri("https://dl.bintray.com/kotlin/kotlinx")
-        }
-        maven {
-            name = "kotlin-eap"
-            url = uri("https://dl.bintray.com/kotlin/kotlin-eap")
-        }
     }
 }
 
@@ -221,22 +206,3 @@
         }
     }
 }
-
-fun Project.applyKotlinProjectConventions() {
-    apply(plugin = "org.gradle.kotlin.kotlin-dsl")
-    apply(plugin = "org.gradle.kotlin-dsl.ktlint-convention")
-
-    plugins.withType<KotlinDslPlugin> {
-        configure<KotlinDslPluginOptions> {
-            experimentalWarning.set(false)
-        }
-    }
-
-    configure<org.jlleitschuh.gradle.ktlint.KtlintExtension> {
-        // TODO:kotlin-dsl remove precompiled script plugins accessors exclusion from ktlint checks
-        filter {
-            exclude("gradle/kotlin/dsl/accessors/_*/**")
-        }
-    }
-}
-
diff -urN gradle-6.0.1-orig/buildSrc/settings.gradle.kts gradle-6.0.1/buildSrc/settings.gradle.kts
--- gradle-6.0.1-orig/buildSrc/settings.gradle.kts	1980-02-01 00:00:00.000000000 +0300
+++ gradle-6.0.1/buildSrc/settings.gradle.kts	2019-12-16 17:14:15.000000000 +0300
@@ -16,10 +16,6 @@
 
 pluginManagement {
     repositories {
-        maven {
-            name = "kotlin-eap"
-            url = uri("https://dl.bintray.com/kotlin/kotlin-eap")
-        }
         gradlePluginPortal()
     }
 }
@@ -42,7 +38,6 @@
 include("docs")
 include("ide")
 include("integrationTesting")
-include("kotlinDsl")
 include("uberPlugins")
 include("packaging")
 include("plugins")
diff -urN gradle-6.0.1-orig/settings.gradle.kts gradle-6.0.1/settings.gradle.kts
--- gradle-6.0.1-orig/settings.gradle.kts	1980-02-01 00:00:00.000000000 +0300
+++ gradle-6.0.1/settings.gradle.kts	2019-12-16 17:05:41.000000000 +0300
@@ -128,14 +128,6 @@
 include("buildCachePackaging")
 include("execution")
 include("buildProfile")
-include("kotlinCompilerEmbeddable")
-include("kotlinDsl")
-include("kotlinDslProviderPlugins")
-include("kotlinDslPlugins")
-include("kotlinDslToolingModels")
-include("kotlinDslToolingBuilders")
-include("kotlinDslTestFixtures")
-include("kotlinDslIntegTests")
 include("workerProcesses")
 include("pineapple")
 include("samples")
