--- xl2tpd-1.3.1-orig/control.c	2011-10-06 21:22:05.000000000 +0200
+++ xl2tpd-1.3.1/control.c	2013-03-30 21:45:04.641605756 +0100
@@ -898,7 +898,7 @@
                 po = add_opt (po, c->lac->pppoptfile);
             }
         };
-	po = add_opt (po, "ipparam");
+	po = add_opt (po, "remotenumber");
         po = add_opt (po, IPADDY (t->peer.sin_addr));
         start_pppd (c, po);
         opt_destroy (po);
@@ -974,7 +974,7 @@
             po = add_opt (po, "file");
             po = add_opt (po, c->lns->pppoptfile);
         }
-	po = add_opt (po, "ipparam");
+	po = add_opt (po, "remotenumber");
         po = add_opt (po, IPADDY (t->peer.sin_addr));
         start_pppd (c, po);
         opt_destroy (po);
@@ -1034,7 +1034,7 @@
                 po = add_opt (po, c->lac->pppoptfile);
             }
         };
-	po = add_opt (po, "ipparam");
+	po = add_opt (po, "remotenumber");
         po = add_opt (po, IPADDY (t->peer.sin_addr));
         start_pppd (c, po);
 
