diff --color -urN goaccess-1.9.2-orig/config/goaccess.conf goaccess-1.9.2/config/goaccess.conf
--- goaccess-1.9.2-orig/config/goaccess.conf	2024-04-09 02:11:20.000000000 +0300
+++ goaccess-1.9.2/config/goaccess.conf	2025-08-06 16:10:26.000000000 +0300
@@ -118,6 +118,20 @@
 # Traefik CLF flavor
 #log-format %h - %e [%d:%t %^] "%r" %s %b "%R" "%u" %^ "%v" "%U" %Lms
 
+# WEBKAOS basic formats
+#time-format %T
+#date-format %d/%b/%Y
+# -- WEBKAOS | main
+#log-format [%^] %h - %e [%d:%t %^] "%r" %s %b "%R" "%u" "%~h"
+# -- WEBKAOS | extended
+#log-format [%^] %h - [%d:%t %^] "%r" %s %b "%~h" "%R" %v %T %^ %^ - %^
+# -- WEBKAOS | reflog
+#log-format [%^] %h - %e [%d:%t %^] "%r" %s %b "%R" "%u"
+# -- WEBKAOS | timed_combined
+#log-format [%^] %h - %e [%d:%t %^] "%r" %s %b "%^" %v "%R" "%u" %T %^
+# -- WEBKAOS | vhost_ip_full_format
+#log-format [%^] %h - %e [%d:%t %^] %v %^ %r %s %b "%R" "%u" "%~h" %T %^
+
 # In addition to specifying the raw log/date/time formats, for
 # simplicity, any of the following predefined log format names can be
 # supplied to the log/date/time-format variables. GoAccess  can  also
@@ -459,7 +473,7 @@
 # See config/browsers.list for an example or
 # https://raw.githubusercontent.com/allinurl/goaccess/master/config/browsers.list
 #
-#browsers-file <filename>
+browsers-file /etc/goaccess/browsers.list
 
 # Date specificity. Possible values: `date` (default), or `hr` or `min`.
 #
@@ -741,7 +755,7 @@
 # Download the GeoLite2-Country.mmdb.gz
 # gunzip GeoLite2-Country.mmdb.gz
 #
-#geoip-database /usr/local/share/GeoIP/GeoLiteCity.dat
+#geoip-database /etc/goaccess/geo.mmdb
 
 ######################################
 # Persistence Options
