--- lib/multi.c
+++ lib/multi.c
@@ -1066,9 +1066,16 @@ static CURLMcode multi_runsingle(struct
         if(result)
           disconnect_conn = TRUE;
         else {
-          if(async)
+          if(async) {
             /* We're now waiting for an asynchronous name lookup */
             multistate(data, CURLM_STATE_WAITRESOLVE);
+#ifdef USE_THREADS_POSIX
+            /* Curl_resolv_getsock() is not properly implemented in case
+             * we use POSIX threaded DNS resolver, we have to hang */
+            result = CURLM_CALL_MULTI_PERFORM;
+            break;
+#endif
+          }
           else {
             /* after the connect has been sent off, go WAITCONNECT unless the
                protocol connect is already done and we can go directly to
@@ -1158,6 +1165,14 @@ static CURLMcode multi_runsingle(struct
         disconnect_conn = TRUE;
         break;
       }
+#ifdef USE_THREADS_POSIX
+      /* Curl_resolv_getsock() is not properly implemented yet in case
+       * we use POSIX threaded DNS resolver, we have to hang */
+      Curl_socket_ready(CURL_SOCKET_BAD, CURL_SOCKET_BAD, 100 /* ms */);
+      result = CURLM_CALL_MULTI_PERFORM;
+      break;
+#endif
+
     }
     break;
 
