diff --color -urN redis-6.0.20-orig/sentinel.conf redis-6.0.20-sentinel/sentinel.conf
--- redis-6.0.20-orig/sentinel.conf	2023-07-10 14:37:46.000000000 +0300
+++ redis-6.0.20-sentinel/sentinel.conf	2023-08-16 11:16:34.000000000 +0300
@@ -1,5 +1,3 @@
-# Example sentinel.conf
-
 # *** IMPORTANT ***
 #
 # By default Sentinel will not be reachable from interfaces different than
@@ -33,7 +31,7 @@
 # Specify the log file name. Also the empty string can be used to force
 # Sentinel to log on the standard output. Note that if you use standard
 # output for logging but daemonize, logs will be sent to /dev/null
-logfile ""
+logfile /var/log/redis/sentinel.log
 
 # sentinel announce-ip <ip>
 # sentinel announce-port <port>
