diff --color -urN redis-7.4.0-orig/sentinel.conf redis-7.4.0-sentinel/sentinel.conf
--- redis-7.4.0-orig/sentinel.conf	2024-07-29 08:31:59.000000000 +0300
+++ redis-7.4.0-sentinel/sentinel.conf	2024-08-20 17:15:01.000000000 +0300
@@ -1,5 +1,3 @@
-# Example sentinel.conf
-
 # By default protected mode is disabled in sentinel mode. Sentinel is reachable
 # from interfaces different than localhost. Make sure the sentinel instance is
 # protected from the outside world via firewalling or other means.
@@ -17,7 +15,7 @@
 # When running daemonized, Redis Sentinel writes a pid file in
 # /var/run/redis-sentinel.pid by default. You can specify a custom pid file
 # location here.
-pidfile /var/run/redis-sentinel.pid
+pidfile /var/run/redis/sentinel.pid
 
 # Specify the server verbosity level.
 # This can be one of:
@@ -31,7 +29,7 @@
 # Specify the log file name. Also the empty string can be used to force
 # Sentinel to log on the standard output. Note that if you use standard
 # output for logging but daemonize, logs will be sent to /dev/null
-logfile ""
+logfile /var/log/redis/sentinel.log
 
 # To enable logging to the system logger, just set 'syslog-enabled' to yes,
 # and optionally update the other syslog parameters to suit your needs.
