diff -urN redis-6.2.0-orig/sentinel.conf redis-6.2.0/sentinel.conf
--- redis-6.2.0-orig/sentinel.conf	2021-02-23 00:23:58.000000000 +0300
+++ redis-6.2.0/sentinel.conf	2022-05-25 17:46:00.000000000 +0300
@@ -1,5 +1,3 @@
-# Example sentinel.conf
-
 # *** IMPORTANT ***
 #
 # By default Sentinel will not be reachable from interfaces different than
@@ -33,8 +31,8 @@
 # Specify the log file name. Also the empty string can be used to force
 # Sentinel to log on the standard output. Note that if you use standard
 # output for logging but daemonize, logs will be sent to /dev/null
-logfile ""
-
+logfile /var/log/redis/sentinel.log
+  
 # sentinel announce-ip <ip>
 # sentinel announce-port <port>
 #
