diff -urN redis-4.0.11-orig/sentinel.conf redis-4.0.11/sentinel.conf
--- redis-4.0.11-orig/sentinel.conf	2018-08-04 01:44:56.000000000 +0300
+++ redis-4.0.11/sentinel.conf	2018-08-07 15:09:33.000000000 +0300
@@ -1,5 +1,3 @@
-# Example sentinel.conf
-
 # *** IMPORTANT ***
 #
 # By default Sentinel will not be reachable from interfaces different than
@@ -16,10 +14,23 @@
 #
 # protected-mode no
 
+# By default Redis Sentinel does not run as a daemon. Use 'yes' if you need it.
+daemonize yes
+
 # port <sentinel-port>
 # The port that this sentinel instance will run on
 port 26379
 
+# When running daemonized, Redis Sentinelwrites a pid file in
+# /var/run/redis.pid by default. You can specify a custom pid file
+# location here.
+pidfile /var/run/redis/sentinel.pid
+
+# Specify the log file name. Also the empty string can be used to force
+# Redis Sentinel to log on the standard output. Note that if you use standard
+# output for logging but daemonize, logs will be sent to /dev/null
+logfile /var/log/redis/sentinel.log
+
 # sentinel announce-ip <ip>
 # sentinel announce-port <port>
 #
@@ -202,4 +213,3 @@
 # to get the program executed.
 
 sentinel deny-scripts-reconfig yes
-
