diff -urN redis-6.0.0-orig/sentinel.conf redis-6.0.0/sentinel.conf
--- redis-6.0.0-orig/sentinel.conf	2020-04-30 16:04:41.000000000 +0300
+++ redis-6.0.0/sentinel.conf	2020-05-22 15:34:44.000000000 +0300
@@ -1,5 +1,3 @@
-# Example sentinel.conf
-
 # *** IMPORTANT ***
 #
 # By default Sentinel will not be reachable from interfaces different than
@@ -23,7 +21,7 @@
 # By default Redis Sentinel does not run as a daemon. Use 'yes' if you need it.
 # Note that Redis will write a pid file in /var/run/redis-sentinel.pid when
 # daemonized.
-daemonize no
+daemonize yes
 
 # When running daemonized, Redis Sentinel writes a pid file in
 # /var/run/redis-sentinel.pid by default. You can specify a custom pid file
@@ -33,7 +31,7 @@
 # Specify the log file name. Also the empty string can be used to force
 # Sentinel to log on the standard output. Note that if you use standard
 # output for logging but daemonize, logs will be sent to /dev/null
-logfile ""
+logfile /var/log/redis/sentinel.log
 
 # sentinel announce-ip <ip>
 # sentinel announce-port <port>
