diff -ru redis-2.8.8-orig/deps/linenoise/linenoise.c redis-2.8.8/deps/linenoise/linenoise.c
--- redis-2.8.8-orig/deps/linenoise/linenoise.c	2014-04-21 10:25:00.571518233 +0200
+++ redis-2.8.8/deps/linenoise/linenoise.c	2014-04-21 10:32:22.000000000 +0200
@@ -104,6 +104,7 @@
 #include <ctype.h>
 #include <sys/types.h>
 #include <sys/ioctl.h>
+#include <sys/stat.h>
 #include <unistd.h>
 #include "linenoise.h"
 
@@ -1063,6 +1064,7 @@
     for (j = 0; j < history_len; j++)
         fprintf(fp,"%s\n",history[j]);
     fclose(fp);
+    chmod(filename,S_IWUSR|S_IRUSR);
     return 0;
 }
 
