diff -urN redis-3.2.1-orig/sentinel.conf redis-3.2.1/sentinel.conf
--- redis-3.2.1-orig/sentinel.conf	2016-06-18 17:49:12.923649425 -0400
+++ redis-3.2.1/sentinel.conf	2016-06-18 17:55:58.000000000 -0400
@@ -1,5 +1,18 @@
 # Example sentinel.conf
 
+# By default Redis Sentinel does not run as a daemon. Use 'yes' if you need it.
+daemonize yes
+
+# When running daemonized, Redis Sentinelwrites a pid file in 
+# /var/run/redis.pid by default. You can specify a custom pid file 
+# location here.
+pidfile /var/run/redis/sentinel.pid
+
+# Specify the log file name. Also the empty string can be used to force
+# Redis Sentinel to log on the standard output. Note that if you use standard
+# output for logging but daemonize, logs will be sent to /dev/null
+logfile /var/log/redis/sentinel.log
+
 # port <sentinel-port>
 # The port that this sentinel instance will run on
 port 26379
@@ -177,5 +190,3 @@
 # Example:
 #
 # sentinel client-reconfig-script mymaster /var/redis/reconfig.sh
-
-
