diff -urN nano-6.0-orig/doc/sample.nanorc.in nano-6.0/doc/sample.nanorc.in
--- nano-6.0-orig/doc/sample.nanorc.in	2021-11-29 13:35:25.000000000 +0300
+++ nano-6.0/doc/sample.nanorc.in	2022-08-19 12:36:51.000000000 +0300
@@ -62,13 +62,13 @@
 # set historylog
 
 ## Display a "scrollbar" on the righthand side of the edit window.
-# set indicator
+set indicator
 
 ## Scroll the buffer contents per half-screen instead of per line.
 # set jumpyscrolling
 
 ## Display line numbers to the left (and any anchors in the margin).
-# set linenumbers
+set linenumbers
 
 ## Enable vim-style lock-files.  This is just to let a vim user know you
 ## are editing a file [s]he is trying to edit and vice versa.  There are
@@ -81,7 +81,7 @@
 ## The opening and closing brackets that are found by a matching-bracket
 ## search.  This may not contain blank characters.  The opening brackets
 ## must come before the closing ones, and they must be in the same order.
-# set matchbrackets "(<[{)>]}"
+set matchbrackets "(<[{)>]}"
 
 ## Suppress the title bar and show the filename plus a cursor-position
 ## percentage in the space of the status bar.
@@ -157,18 +157,18 @@
 # set smarthome
 
 ## Spread overlong lines over multiple screen lines.
-# set softwrap
+set softwrap
 
 ## Use this spelling checker instead of the internal one.  This option
 ## does not have a default value.
-# set speller "aspell -x -c"
+set speller "hunspell"
 
 ## Use the end of the title bar for some state flags: I = auto-indenting,
 ## M = mark, L = hard-wrapping long lines, R = recording, S = soft-wrapping.
 # set stateflags
 
 ## Use this tab size instead of the default; it must be greater than 0.
-# set tabsize 8
+set tabsize 2
 
 ## Convert each typed tab to the fitting number of spaces.
 # set tabstospaces
@@ -228,6 +228,12 @@
 # set keycolor lightmagenta
 # set functioncolor magenta
 
+set titlecolor brightwhite,blue
+set statuscolor brightwhite,green
+set errorcolor brightwhite,red
+set selectedcolor brightwhite,magenta
+set numbercolor brightblack
+set scrollercolor brightblack
 
 ## === Syntax coloring ===
 ## For all details, see 'man nanorc', section SYNTAX HIGHLIGHTING.
@@ -235,6 +241,8 @@
 ## To include most of the existing syntax definitions, you can do:
 # include "@PKGDATADIR@/*.nanorc"
 
+include "/usr/share/nano/*.nanorc"
+
 ## Or you can select just the ones you need.  For example:
 # include "@PKGDATADIR@/html.nanorc"
 # include "@PKGDATADIR@/python.nanorc"
