diff -urN nano-4.0-orig/doc/sample.nanorc.in nano-4.0/doc/sample.nanorc.in
--- nano-4.0-orig/doc/sample.nanorc.in	2019-03-09 12:21:52.000000000 +0300
+++ nano-4.0/doc/sample.nanorc.in	2019-08-17 14:38:24.000000000 +0300
@@ -71,7 +71,7 @@
 # set jumpyscrolling
 
 ## Display line numbers to the left of the text.
-# set linenumbers
+set linenumbers
 
 ## Enable vim-style lock-files.  This is just to let a vim user know you
 ## are editing a file [s]he is trying to edit and vice versa.  There are
@@ -151,17 +151,17 @@
 # set smarthome
 
 ## Enable soft line wrapping (AKA full-line display).
-# set softwrap
+set softwrap
 
 ## Use this spelling checker instead of the internal one.  This option
 ## does not have a default value.
-# set speller "aspell -x -c"
+set speller "hunspell"
 
 ## Allow nano to be suspended.
 # set suspend
 
 ## Use this tab size instead of the default; it must be greater than 0.
-# set tabsize 8
+set tabsize 2
 
 ## Convert typed tabs to spaces.
 # set tabstospaces
@@ -216,6 +216,11 @@
 # set keycolor brightmagenta
 # set functioncolor magenta
 
+set titlecolor brightwhite,blue
+set statuscolor brightwhite,green
+set errorcolor brightwhite,red
+set selectedcolor brightwhite,magenta
+set numbercolor brightblack
 
 ## Setup of syntax coloring.
 ##
@@ -259,7 +264,7 @@
 ## them easier to remember and faster to type using nano's -Y option.
 
 ## To include all existing syntax definitions, you can do:
-# include "@PKGDATADIR@/*.nanorc"
+include "/usr/share/nano/*.nanorc"
 
 
 ## Key bindings.
