diff -urN nano-4.5-orig/doc/sample.nanorc.in nano-4.5/doc/sample.nanorc.in
--- nano-4.5-orig/doc/sample.nanorc.in	2019-10-04 12:38:05.000000000 +0300
+++ nano-4.5/doc/sample.nanorc.in	2020-01-17 14:59:09.000000000 +0300
@@ -66,7 +66,7 @@
 # set jumpyscrolling
 
 ## Display line numbers to the left of the text.
-# set linenumbers
+set linenumbers
 
 ## Enable vim-style lock-files.  This is just to let a vim user know you
 ## are editing a file [s]he is trying to edit and vice versa.  There are
@@ -76,7 +76,7 @@
 ## The opening and closing brackets that can be found by bracket
 ## searches.  They cannot contain blank characters.  The former set must
 ## come before the latter set, and both must be in the same order.
-# set matchbrackets "(<[{)>]}"
+set matchbrackets "(<[{)>]}"
 
 ## Enable mouse support, if available for your system.  When enabled,
 ## mouse clicks can be used to place the cursor, set the mark (with a
@@ -150,17 +150,17 @@
 # set smarthome
 
 ## Enable soft line wrapping (AKA full-line display).
-# set softwrap
+set softwrap
 
 ## Use this spelling checker instead of the internal one.  This option
 ## does not have a default value.
-# set speller "aspell -x -c"
+set speller "hunspell"
 
 ## Allow nano to be suspended.
 # set suspend
 
 ## Use this tab size instead of the default; it must be greater than 0.
-# set tabsize 8
+set tabsize 2
 
 ## Convert typed tabs to spaces.
 # set tabstospaces
@@ -215,6 +215,11 @@
 # set keycolor brightmagenta
 # set functioncolor magenta
 
+set titlecolor brightwhite,blue
+set statuscolor brightwhite,green
+set errorcolor brightwhite,red
+set selectedcolor brightwhite,magenta
+set numbercolor brightblack
 
 ## Setup of syntax coloring.
 ##
@@ -259,6 +264,7 @@
 
 ## To include all existing syntax definitions, you can do:
 # include "@PKGDATADIR@/*.nanorc"
+include "/usr/share/nano/*.nanorc"
 
 
 ## If <Tab> should always produce four spaces when editing a Python file,
