diff --color -urN nano-8.2-orig/doc/sample.nanorc.in nano-8.2/doc/sample.nanorc.in
--- nano-8.2-orig/doc/sample.nanorc.in	2024-09-05 12:29:01.000000000 +0300
+++ nano-8.2/doc/sample.nanorc.in	2025-01-25 22:05:25.000000000 +0300
@@ -34,7 +34,7 @@
 # set backupdir ""
 
 ## Use bold text instead of reverse video text.
-# set boldtext
+set boldtext
 
 ## Treat any line with leading whitespace as the beginning of a paragraph.
 # set bookstyle
@@ -76,13 +76,13 @@
 # set historylog
 
 ## Display a "scrollbar" on the righthand side of the edit window.
-# set indicator
+set indicator
 
 ## Scroll the buffer contents per half-screen instead of per line.
 # set jumpyscrolling
 
 ## Display line numbers to the left (and any anchors in the margin).
-# set linenumbers
+set linenumbers
 
 ## Enable vim-style lock-files.  This is just to let a vim user know you
 ## are editing a file [s]he is trying to edit and vice versa.  There are
@@ -95,7 +95,7 @@
 ## The opening and closing brackets that are found by a matching-bracket
 ## search.  This may not contain blank characters.  The opening brackets
 ## must come before the closing ones, and they must be in the same order.
-# set matchbrackets "(<[{)>]}"
+set matchbrackets "(<[{)>]}"
 
 ## Suppress the title bar and show the filename plus a cursor-position
 ## percentage in the space of the status bar.
@@ -170,18 +170,18 @@
 # set smarthome
 
 ## Spread overlong lines over multiple screen lines.
-# set softwrap
+set softwrap
 
 ## Use this spelling checker instead of the internal one.  This option
 ## does not have a default value.
-# set speller "aspell -x -c"
+set speller "hunspell"
 
 ## Use the end of the title bar for some state flags: I = auto-indenting,
 ## M = mark, L = hard-wrapping long lines, R = recording, S = soft-wrapping.
 # set stateflags
 
 ## Use this tab size instead of the default; it must be greater than 0.
-# set tabsize 8
+set tabsize 2
 
 ## Convert each typed tab to the fitting number of spaces.
 # set tabstospaces
@@ -244,6 +244,14 @@
 # set keycolor lightmagenta
 # set functioncolor magenta
 
+set titlecolor normal,blue
+set statuscolor normal,green
+set errorcolor normal,red
+set selectedcolor white,magenta
+set numbercolor brightblack
+set scrollercolor lightwhite
+set keycolor lightwhite
+set functioncolor lightwhite
 
 ## === Syntax coloring ===
 ## For all details, see 'man nanorc', section SYNTAX HIGHLIGHTING.
@@ -256,6 +264,8 @@
 # include @PKGDATADIR@/python.nanorc
 # include @PKGDATADIR@/sh.nanorc
 
+include "/usr/share/nano/*.nanorc"
+
 ## In @PKGDATADIR@/extra/ you can find some syntaxes that are
 ## specific for certain distros or for some less common languages.
 
