diff -urN salt-2018.3.1-orig/conf/cloud salt-2018.3.1/conf/cloud
--- salt-2018.3.1-orig/conf/cloud	2018-06-13 19:03:06.000000000 +0300
+++ salt-2018.3.1/conf/cloud	2018-06-19 15:53:59.000000000 +0300
@@ -22,7 +22,7 @@
 
 # The location of the master log file
 #
-#log_file: /var/log/salt/cloud
+log_file: /var/log/salt/cloud.log
 
 
 # The level of messages to send to the console.
diff -urN salt-2018.3.1-orig/conf/master salt-2018.3.1/conf/master
--- salt-2018.3.1-orig/conf/master	2018-06-13 19:03:06.000000000 +0300
+++ salt-2018.3.1/conf/master	2018-06-19 15:57:13.000000000 +0300
@@ -9,7 +9,7 @@
 # Per default, the master will automatically include all config files
 # from master.d/*.conf (master.d is a directory in the same directory
 # as the main master config file).
-#default_include: master.d/*.conf
+default_include: master.d/*.conf
 
 # The address of the interface to bind to:
 #interface: 0.0.0.0
@@ -32,7 +32,7 @@
 #ret_port: 4506
 
 # Specify the location of the daemon process ID file:
-#pidfile: /var/run/salt-master.pid
+pidfile: /var/run/salt-master.pid
 
 # The root directory prepended to these options: pki_dir, cachedir,
 # sock_dir, log_file, autosign_file, autoreject_file, extension_modules,
@@ -43,7 +43,7 @@
 #conf_file: /etc/salt/master
 
 # Directory used to store public key data:
-#pki_dir: /etc/salt/pki/master
+pki_dir: /etc/salt/pki/master
 
 # Key cache. Increases master speed for large numbers of accepted
 # keys. Available options: 'sched'. (Updates on a fixed schedule.)
@@ -55,7 +55,7 @@
 # Directory to store job and cache data:
 # This directory may contain sensitive data and should be protected accordingly.
 #
-#cachedir: /var/cache/salt/master
+cachedir: /var/cache/salt/master
 
 # Directory for custom modules. This directory can contain subdirectories for
 # each of Salt's module types such as "runners", "output", "wheel", "modules",
@@ -120,7 +120,7 @@
 #cli_summary: False
 
 # Set the directory used to hold unix sockets:
-#sock_dir: /var/run/salt/master
+sock_dir: /var/run/salt/master
 
 # The master can take a while to start up when lspci and/or dmidecode is used
 # to populate the grains for the master. Enable if you want to see GPU hardware
@@ -514,7 +514,7 @@
 #ssh_user: root
 
 # The log file of the salt-ssh command:
-#ssh_log_file: /var/log/salt/ssh
+ssh_log_file: /var/log/salt/ssh.log
 
 # Pass in minion option overrides that will be inserted into the SHIM for
 # salt-ssh calls. The local minion config is not used for salt-ssh. Can be
@@ -705,7 +705,7 @@
 #
 # Prior to changing this value, the master should be stopped and all Salt
 # caches should be cleared.
-#hash_type: sha256
+hash_type: sha256
 
 # The buffer size in the file server can be adjusted here:
 #file_buffer_size: 1048576
@@ -1061,10 +1061,10 @@
 #syndic_master_port: 4506
 
 # PID file of the syndic daemon:
-#syndic_pidfile: /var/run/salt-syndic.pid
+syndic_pidfile: /var/run/salt-syndic.pid
 
 # The log file of the salt-syndic daemon:
-#syndic_log_file: /var/log/salt/syndic
+syndic_log_file: /var/log/salt/syndic.log
 
 # The behaviour of the multi-syndic when connection to a master of masters failed.
 # Can specify ``random`` (default) or ``ordered``. If set to ``random``, masters
@@ -1149,8 +1149,8 @@
 #log_file: file:///dev/log
 #log_file: udp://loghost:10514
 
-#log_file: /var/log/salt/master
-#key_logfile: /var/log/salt/key
+log_file: /var/log/salt/master.log
+key_logfile: /var/log/salt/key.log
 
 # The level of messages to send to the console.
 # One of 'garbage', 'trace', 'debug', info', 'warning', 'error', 'critical'.
diff -urN salt-2018.3.1-orig/conf/minion salt-2018.3.1/conf/minion
--- salt-2018.3.1-orig/conf/minion	2018-06-13 19:03:06.000000000 +0300
+++ salt-2018.3.1/conf/minion	2018-06-19 15:59:10.000000000 +0300
@@ -9,7 +9,7 @@
 # Per default the minion will automatically include all config files
 # from minion.d/*.conf (minion.d is a directory in the same directory
 # as the main minion config file).
-#default_include: minion.d/*.conf
+default_include: minion.d/*.conf

 # Set the location of the salt master server. If the master server cannot be
 # resolved, then the minion will fail to start.
@@ -92,7 +92,7 @@
 #sudo_user: root

 # Specify the location of the daemon process ID file.
-#pidfile: /var/run/salt-minion.pid
+pidfile: /var/run/salt-minion.pid

 # The root directory prepended to these options: pki_dir, cachedir, log_file,
 # sock_dir, pidfile.
@@ -102,7 +102,7 @@
 #conf_file: /etc/salt/minion

 # The directory to store the pki information in
-#pki_dir: /etc/salt/pki/minion
+pki_dir: /etc/salt/pki/minion

 # Explicitly declare the id for this minion to use, if left commented the id
 # will be the hostname as returned by the python call: socket.getfqdn()
@@ -136,7 +136,7 @@
 #
 # Where cache data goes.
 # This data may contain sensitive data and should be protected accordingly.
-#cachedir: /var/cache/salt/minion
+cachedir: /var/cache/salt/minion

 # Append minion_id to these directories.  Helps with
 # multiple proxies and minions running on the same machine.
@@ -155,7 +155,7 @@
 #cache_jobs: False

 # Set the directory used to hold unix sockets.
-#sock_dir: /var/run/salt/minion
+sock_dir: /var/run/salt/minion

 # The minion can take a while to start up when lspci and/or dmidecode is used
 # to populate the grains for the minion. Set this to False if you do not need
@@ -621,7 +621,7 @@
 #
 # Warning: Prior to changing this value, the minion should be stopped and all
 # Salt caches should be cleared.
-#hash_type: sha256
+hash_type: sha256

 # The Salt pillar is searched for locally if file_client is set to local. If
 # this is the case, and pillar data is defined, then the pillar_roots need to
@@ -739,8 +739,8 @@
 #log_file: file:///dev/log
 #log_file: udp://loghost:10514
 #
-#log_file: /var/log/salt/minion
-#key_logfile: /var/log/salt/key
+log_file: /var/log/salt/minion.log
+key_logfile: /var/log/salt/key.log

 # The level of messages to send to the console.
 # One of 'garbage', 'trace', 'debug', 'info', 'warning', 'error', 'critical'.

diff -urN salt-2018.3.1-orig/conf/proxy salt-2018.3.1/conf/proxy
--- salt-2018.3.1-orig/conf/proxy	2018-06-13 19:03:06.000000000 +0300
+++ salt-2018.3.1/conf/proxy	2018-06-19 15:59:45.000000000 +0300
@@ -538,8 +538,8 @@
 #log_file: file:///dev/log
 #log_file: udp://loghost:10514
 #
-#log_file: /var/log/salt/minion
-#key_logfile: /var/log/salt/key
+log_file: /var/log/salt/minion.log
+key_logfile: /var/log/salt/key.log

 # The level of messages to send to the console.
 # One of 'garbage', 'trace', 'debug', 'info', 'warning', 'error', 'critical'.

