--- etc/pgbouncer.ini.old	2012-04-24 01:42:43.093199723 +0300
+++ etc/pgbouncer.ini	2012-04-24 01:43:58.083201432 +0300
@@ -59,10 +59,10 @@
 ;;;
 
 ; comma-separated list of users, who are allowed to change settings
-;admin_users = user2, someadmin, otheradmin
+admin_users = postgres
 
 ; comma-separated list of users who are just allowed to use SHOW command
-;stats_users = stats, root
+stats_users = stats, postgres
 
 ;;;
 ;;; Pooler personality questions
