diff -ur gdal-1.8.0/swig/perl/GNUmakefile gdal-1.8.0_fix//swig/perl/GNUmakefile
--- gdal-1.8.0/swig/perl/GNUmakefile	2011-01-14 06:19:06.000000000 +0100
+++ gdal-1.8.0_fix//swig/perl/GNUmakefile	2011-02-04 13:37:06.000000000 +0100
@@ -26,7 +26,7 @@
 gdal_wrap.cc: generate
 
 Makefile_Geo__GDAL:
-	perl Makefile.PL INSTALL_BASE=$(INST_PREFIX)
+	perl Makefile.PL INSTALLDIRS=vendor
 
 test: build
 	$(MAKE) -f Makefile_Geo__GDAL test
