--- gdal-1.5.4/swig/python/GNUmakefile.bak	2008-07-26 21:04:24.000000000 +0200
+++ gdal-1.5.4/swig/python/GNUmakefile	2009-05-18 20:14:35.000000000 +0200
@@ -53,12 +53,7 @@
 	$(PYTHON) setup.py bdist_egg 
 	
 install:
-
-ifeq ($(PY_HAVE_SETUPTOOLS),1)
-	$(PYTHON) setup.py install 
-else
-	$(PYTHON) setup.py install --prefix=$(DESTDIR)$(prefix)
-endif
+	$(PYTHON) setup.py install --prefix=$(prefix) --root=$(DESTDIR)
 
 	for f in $(SCRIPTS) ; do $(INSTALL) ./scripts/$$f $(DESTDIR)$(INST_BIN) ; done

