diff -urN pg_comparator-2.3.1-orig/Makefile pg_comparator-2.3.1/Makefile
--- pg_comparator-2.3.1-orig/Makefile	2017-09-18 00:42:12.653308101 +0200
+++ pg_comparator-2.3.1/Makefile	2017-09-18 00:42:46.000000000 +0200
@@ -16,7 +16,7 @@
 EXTRA_CLEAN	= $(name).1 $(name).html pod2htm?.tmp $(EXTENSION).control
 
 # get postgresql extension infrastructure
-PG_CONFIG	= pg_config
+PG_CONFIG	?= pg_config
 PGXS	:= $(shell $(PG_CONFIG) --pgxs)
 include $(PGXS)
 
