diff --git a/file.c b/file.c
index 289aee6..d710e30 100644
--- a/file.c
+++ b/file.c
@@ -1229,7 +1229,7 @@ int parse_config (FILE * f)
     /* Read in the configuration file handed to us */
     /* FIXME: I should check for incompatible options */
     int context = 0;
-    char buf[STRLEN];
+    char buf[1024]; 
     char *s, *d, *t;
     int linenum = 0;
     int def = 0;
@@ -1395,7 +1395,7 @@ int parse_config (FILE * f)
             }
             if (!(t = strchr (s, '=')))
             {
-                l2tp_log (LOG_WARNING, "parse_config: line %d: no '=' in data\n",
+                l2tp_log (LOG_WARNING, "parse_config: line %d: line too long or no '=' in data\n",
                      linenum);
                 return -1;
             }
