diff -Naur xl2tpd-1.3.1-orig/Makefile xl2tpd-1.3.1/Makefile
--- xl2tpd-1.3.1-orig/Makefile	2013-04-01 16:05:42.459122589 -0400
+++ xl2tpd-1.3.1/Makefile	2013-04-01 16:06:03.800359414 -0400
@@ -98,6 +98,7 @@
 #LIBS= $(OSLIBS) # -lefence # efence for malloc checking
 EXEC=xl2tpd
 CONTROL_EXEC=xl2tpd-control
+CONTROL_OBJ=xl2tpd-control.o
 
 PREFIX?=/usr/local
 SBINDIR?=$(DESTDIR)${PREFIX}/sbin
@@ -114,7 +115,8 @@
 	$(CC) $(LDFLAGS) -o $@ $(OBJS) $(LDLIBS)
 
 $(CONTROL_EXEC): $(CONTROL_SRCS)
-	$(CC) $(CONTROL_SRCS) -o $@
+	$(CC) $(CFLAGS) -c $(CONTROL_SRCS)
+	$(CC) $(LDFLAGS) -o $@ $(CONTROL_OBJ)
 
 pfc:
 	$(CC) $(CFLAGS) -c contrib/pfc.c
