diff -urN salt-2015.8.8-orig/salt/config.py salt-2015.8.8/salt/config.py
--- salt-2015.8.8-orig/salt/config.py 2016-03-23 01:55:42.000000000 +0300
+++ salt-2015.8.8/salt/config.py  2016-04-06 10:49:40.581725971 +0300
@@ -923,8 +923,6 @@
     'recon_randomize': True,
     'return_retry_timer': 5,
     'return_retry_timer_max': 10,
-    'syndic_log_file': os.path.join(salt.syspaths.LOGS_DIR, 'syndic'),
-    'syndic_pidfile': os.path.join(salt.syspaths.PIDFILE_DIR, 'salt-syndic.pid'),
     'random_reauth_delay': 10,
     'winrepo_source_dir': 'salt://win/repo-ng/',
     'winrepo_dir': os.path.join(salt.syspaths.BASE_FILE_ROOTS_DIR, 'win', 'repo'),
@@ -1081,6 +1079,8 @@
     'peer': {},
     'preserve_minion_cache': False,
     'syndic_master': '',
+    'syndic_log_file': os.path.join(salt.syspaths.LOGS_DIR, 'syndic'),
+    'syndic_pidfile': os.path.join(salt.syspaths.PIDFILE_DIR, 'salt-syndic.pid'),
     'runner_dirs': [],
     'outputter_dirs': [],
     'client_acl': {},
@@ -1677,8 +1677,8 @@
     minion_opts = minion_config(
         minion_config_path, minion_env_var, minion_defaults
     )
-    opts['_minion_conf_file'] = master_opts['conf_file']
-    opts['_master_conf_file'] = minion_opts['conf_file']
+    opts['_minion_conf_file'] = minion_opts['conf_file']
+    opts['_master_conf_file'] = master_opts['conf_file']
     opts.update(master_opts)
     opts.update(minion_opts)
     syndic_opts = {
