diff -urN go-orig/src/make.bash go/src/make.bash
--- go-orig/src/make.bash	2015-09-01 23:20:11.476199893 +0200
+++ go/src/make.bash	2015-09-01 23:27:56.000000000 +0200
@@ -113,12 +113,13 @@
 echo cmd/dist
 export GOROOT="$(cd .. && pwd)"
 GOROOT_BOOTSTRAP=${GOROOT_BOOTSTRAP:-$HOME/go1.4}
-if [ ! -x "$GOROOT_BOOTSTRAP/bin/go" ]; then
+GOROOT_BOOTSTRAP_BIN=$(find $GOROOT_BOOTSTRAP/bin -name 'go' | head -1)
+if [ ! -x "$GOROOT_BOOTSTRAP_BIN" ]; then
 	echo "ERROR: Cannot find $GOROOT_BOOTSTRAP/bin/go." >&2
 	echo "Set \$GOROOT_BOOTSTRAP to a working Go tree >= Go 1.4." >&2
 fi
 rm -f cmd/dist/dist
-GOROOT="$GOROOT_BOOTSTRAP" GOOS="" GOARCH="" "$GOROOT_BOOTSTRAP/bin/go" build -o cmd/dist/dist ./cmd/dist
+GOROOT="$GOROOT_BOOTSTRAP" GOOS="" GOARCH="" "$GOROOT_BOOTSTRAP_BIN" build -o cmd/dist/dist ./cmd/dist
 
 # -e doesn't propagate out of eval, so check success by hand.
 eval $(./cmd/dist/dist env -p || echo FAIL=true)
