diff -urN postgis-2.4.5-orig/raster/rt_pg/Makefile.in postgis-2.4.5/raster/rt_pg/Makefile.in
--- postgis-2.4.5-orig/raster/rt_pg/Makefile.in	2018-09-12 19:46:37.000000000 +0300
+++ postgis-2.4.5/raster/rt_pg/Makefile.in	2018-11-18 02:30:14.000000000 +0300
@@ -66,7 +66,7 @@
 LIBPROJ_CFLAGS=@PROJ_CPPFLAGS@
 
 override CFLAGS := $(LIBLWGEOM_CFLAGS) @CFLAGS@
-PG_CPPFLAGS+= @CPPFLAGS@ $(LIBGDAL_CFLAGS) $(LIBPGCOMMON_CFLAGS) $(LIBPROJ_CFLAGS) -I../rt_core
+	PG_CPPFLAGS+= @CPPFLAGS@ $(LIBGDAL_CFLAGS) $(LIBLWGEOM_CFLAGS) $(LIBPGCOMMON_CFLAGS) $(LIBPROJ_CFLAGS) -I../rt_core
 SHLIB_LINK_F = ../rt_core/librtcore.a $(LIBLWGEOM_LDFLAGS) $(LIBPGCOMMON_LDFLAGS) $(LIBGDAL_LDFLAGS) @SHLIB_LINK@
 
 # Extra files to remove during 'make clean'
