diff --color -urN rpmlint-2.4.0-orig/configs/Fedora/fedora.toml rpmlint-2.4.0/configs/Fedora/fedora.toml
--- rpmlint-2.4.0-orig/configs/Fedora/fedora.toml	2022-10-03 17:00:16.000000000 +0300
+++ rpmlint-2.4.0/configs/Fedora/fedora.toml	2023-07-03 20:17:21.000000000 +0300
@@ -39,7 +39,6 @@
 
 # Enabled checks for the rpmlint to be run (besides the default set)
 Checks = [
-    "BashismsCheck",
     "PAMModulesCheck",
     "TmpFilesCheck",
     "SysVInitOnSystemdCheck",
@@ -90,6 +89,7 @@
 #    ' dir-or-file-in-usr-local ',  # handled by CheckFilelist.py
     ' non-standard-dir-in-usr ',  # handled by CheckFilelist.py
     ' no-signature',
+    ' superfluous-%clean-section',
 #    ' symlink-crontab-file',  #bnc591431
 #    ' without-chkconfig',
 #    'unstripped-binary-or-object.*\.ko',
@@ -294,6 +294,8 @@
     'invalid-url .*\.jboss\.org/.*HTTP Error 403',
     'invalid-url .*bitbucket\.org/.*HTTP Error 403',
     'invalid-url .*github\.com/.*HTTP Error 403',
+    ## Ignore local packages
+    'invalid-url Source.* [^\/]*',
     # Don't care about long descriptions on debuginfo packages
     # They automatically include the package name and are always
     # quite long.
